HerPortal.Data/HerDbContext.cs

```diff
@@ -28,6 +28,22 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             .Entity<LocalAuthority>()
             .HasIndex(la => la.CustodianCode)
             .IsUnique();
+
+        modelBuilder
+            .Entity<AuditDownload>()
+            .HasKey(cf => new
```

Comment:
This should work because no-one can hammer the download fast enough to get the same timestamp, but it seems a bit odd to me. I'd go for an explicit auto generated id column for the primary key.